Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -131,7 +131,8 @@ extension BrowserViewController: WKNavigationDelegate {
     }
 
     if InternalURL.isValid(url: url) {
-      if navigationAction.navigationType != .backForward, navigationAction.isInternalUnprivileged {
+      if navigationAction.navigationType != .backForward, navigationAction.isInternalUnprivileged,
+          (navigationAction.sourceFrame != nil || navigationAction.targetFrame?.isMainFrame == false || navigationAction.request.cachePolicy == .useProtocolCachePolicy) {
```

Comment:
I understand we need cache policy to check WebKit is making a request from interactionState vs. a regular request by the user. But Why do we need to check mainFrame here ?

Reply:
Webkit only makes requests from the main-frame. It's just an additional check.
So do our scripts btw. We don't have any scripts that should ever make internal requests from a frame.